### ISSUE TITLE
build replaces files with same name

### ISSUE THREAD

**Issue by alex116** [ThrowTheSwitch/Ceedling#343]
My project does not have a flat directory structure.
`/src/helpers/strings/utils.c`
`/src/helpers/strings/utils.h`
`/src/helpers/time/utils.c`
`/src/helpers/time/utils.h`

When I use both 
`#include "strings/utils.h"`
`#include "time/utils.h"`
in my code I get linker errors because ceedling compiles both to:
`build/test/out/c/utils.o`

can't you just build to 
`build/test/out/c/helpers/time/utils.o`
`build/test/out/c/helpers/strings/utils.o`
and add those folders to the -L flag when linking?

Is there a workaround for this? I don't want to rename my files, my coworkers would not appreciate this.

**Comment by alex116** [comment]
I couldn't let this otherwise good project go to waste so I hacked away at it... I probably did a terrible job and caused many incompatibilities with other setups/configurations but I'm putting ruby on my CV from now on. 

I use:
`#include "strings/mock_utils.h"`
`#include "time/mock_utils.h"`
in my test file.

I now generate the mocks to:
`build/test/mocks/src/helpers/strings/mock_utils.*`
`build/test/mocks/src/helpers/time/mock_utils.*`

I `#include "src/helpers/time/utils.h"` in time/mock_utils.h and
`#include "src/helpers/strings/utils.h"` in strings/mock_utils.h when they get generated. 

They build to 
`build/test/out/c/src/helpers/**/mock_utils.o`

I added something to the project.yml
./. to the src path - because of the new #include in the mock (it's relative to "src")
build/test/mocks/** so we can find the newly generated files in the subfolders

**Comment by IvanRibakov** [comment]
@alex116 any chance you could share the changes you had to make for this to work? Thanks in advance!

**Comment by mvandervoord** [comment]
indeed. this is an often-requested feature. If I don't have to start from scratch, that would be great.

**Comment by alex116** [comment]
I edited my projects' ceedling installation how would I best share this? Uploading the vendor folder would probably be easiest on my part. What do you think?

**Comment by mvandervoord** [comment]
If you have any diff tools on your machine, you could compare your version to the last released version and create a patch file? If not, zipping it up and putting it somewhere for one of us to grab would be great.

**Comment by alex116** [comment]
`git clone --recursive https://github.com/throwtheswitch/ceedling.git
mkdir /tmp/vendor
cp -r ceedling /tmp/vendor/
cp -r myprojectsvendorfolder /tmp/vendor_modified
diff -ruN /tmp/vendor/ceedling/lib/ /tmp/vendor_modified/ceedling/lib/ > mypatch.txt
`

[mypatch.txt](https://github.com/ThrowTheSwitch/Ceedling/files/3372391/mypatch.txt)

**Comment by art-of-dom** [comment]
I can look into making a pull request out of this. I may ping you alex when it's ready.

**Comment by alex116** [comment]
I'm sorry, I forgot to diff CMock. Here it is. Please also consider that I hacked around cmock for this issue: https://github.com/ThrowTheSwitch/CMock/issues/213
[cmockpatch.txt](https://github.com/ThrowTheSwitch/Ceedling/files/3385906/cmockpatch.txt)

**Comment by mkarlesky** [comment]
Sadly, this remains an open issue even 7 years later. Wow. I'm going to close it but reference it in the single issue where all these requests are being collected towards finally resolving this.
